### PR TITLE
Default font set to consolas for windows os

### PR DESCRIPTION
**Pull request by sonulohani**
Default font in windows is by-default set to **monospace** which is incorrect. This PR fixes the default font to consolas

Issue: https://github.com/cpeditor/cpeditor/issues/422 https://github.com/cpeditor/cpeditor/issues/425

### CODE REVIEW

**Comment by coder3101** [comment]
Document your changes to `doc/CHANGELOG.md` under unreleased section, under subsection **Changed**.

**Comment by sonulohani** [comment]
> 
> 
> Document your changes to `doc/CHANGELOG.md` under unreleased section, under subsection **Changed**.

Done.

**Comment by neko-para** [comment]
@sonulohani Can you let the platform feature not restrict to QFont? Maybe other settings need this feature as well.

File: src/Settings/settings.json [hunk 1]
```diff
@@ -16,7 +16,10 @@
     {
         "name": "Editor Font",
         "type": "QFont",
-        "default": "QFont(\"monospace\")",
+        "default": {
+            "Windows": "QFont(\"Consolas\")",
```
Comment (coder3101) [inline src/Settings/settings.json:20]:
```suggestion
            "Windows": "QFont(\"Consolas\", 12)",
```

File: src/Settings/settings.json [hunk 2]
```diff
@@ -700,7 +703,10 @@
     {
         "name": "Test Cases Font",
         "type": "QFont",
-        "default": "QFont(\"monospace\")",
+        "default": {
+            "Windows": "QFont(\"Consolas\")",
```
Comment (coder3101) [inline src/Settings/settings.json:707]:
```suggestion
            "Windows": "QFont(\"Consolas\", 12)",
```

File: src/Settings/settings.json [hunk 3]
```diff
@@ -712,7 +718,10 @@
     {
         "name": "Message Logger Font",
         "type": "QFont",
-        "default": "QFont(\"monospace\")",
+        "default": {
+            "Windows": "QFont(\"Consolas\")",
```
Comment (coder3101) [inline src/Settings/settings.json:722]:
```suggestion
            "Windows": "QFont(\"Consolas\", 12)",
```

**Comment by coder3101** [comment]
> 
> 
> @sonulohani Can you let the platform feature not restrict to QFont? Maybe other settings need this feature as well.

I had already asked for a feature like this in Settings. Platform-specific default is a nice addition.

**Comment by sonulohani** [comment]
> 
> 
> We should use `QFont::TypeWriter` instead of a hardcoded font.

Hi @ouuan,

For now we have settings file as json which has hardcoded values. One thing is that we can remove default font( set "default" as empty under font section) and in code we can set font to default style hint. Please let me know your opinion.

**Comment by neko-para** [comment]
@sonulohani You can use a lambda function to get the right font. The "default" is copied to source code.

**Comment by sonulohani** [comment]
> 
> 
> @sonulohani Can you let the platform feature not restrict to QFont? Maybe other settings need this feature as well.

Done

**Comment by sonulohani** [comment]
> 
> 
> @sonulohani You can use a lambda function to get the right font. The "default" is copied to source code.

Could you please explain.

**Comment by coder3101** [comment]
> For now we have settings file as json which has hardcoded values. One thing is that we can remove default font( set "default" as empty under font section) and in code we can set font to default style hint. Please let me know your opinion.

No, settings.json does not have hardcoded values, it has codes that are expanded in the generated .hpp/.cpp files. As neko said, you can use lambda function like like:

```
default: "[]{/* write some C++ here, return the default QFont */}"

```

It should work.

**Comment by sonulohani** [comment]
> 
> 
> > For now we have settings file as json which has hardcoded values. One thing is that we can remove default font( set "default" as empty under font section) and in code we can set font to default style hint. Please let me know your opinion.
> 
> No, `settings.json` does not have hardcoded values, it has codes that are expanded in the generated .hpp/.cpp files. As neko said, you can use lambda function like like:
> 
> ```
> default: "[]{/* write some C++ here, return the default QFont */}"
> ```
> 
> It should work.

Got it. Thanks.

**Comment by neko-para** [comment]
@sonulohani It should be like "default": "[]() -> QFont { /* some code */ }()"
remember to call it, or the lambda will be stored and no error will appear.

File: doc/CHANGELOG.md [hunk 4]
```diff
@@ -16,10 +16,12 @@ With the portable version, you can easily store it in something like a USB disk,
 ### Fixed
 
 - Fix the wrong default setting for C++ executable file path.
+- Fix Font Size restores to small size after set to default or first launch.
```
Comment (ouuan) [inline doc/CHANGELOG.md:19]: Are you sure this is a "bug"? The font size definitely changes after setting to default. The bug is #422 and nothing else.

Comment (sonulohani) [reply doc/CHANGELOG.md:19]: I am able to reproduce this in current master.

**Comment by ouuan** [comment]
I think we can close this and make a new PR, because these changes won't be used.

**Comment by sonulohani** [comment]
> I think we can close this and make a new PR, because these changes won't be used.

Okay @ouuan .

File: doc/CHANGELOG.md (hunk 4 above)
Comment (coder3101) [reply doc/CHANGELOG.md:19]: > Are you sure this is a "bug"?

Yes, I tested with my 32-inch monitor, it changes to small size in master.

Here is how to reproduce it:
- Change the font to some other say "Consolas"
- apply the new font
- Hit restore to default and apply settings.
- The new default font has a size less than 12 px, something like 8-9.

Comment (coder3101) [reply doc/CHANGELOG.md:19]: image

**Pull request closed without merging**